lib/PlacesProvider.js

```diff
@@ -158,28 +190,117 @@ Links.prototype = {
     // In general the groupby behavior in the absence of aggregates is not
     // defined in SQL, hence we are relying on sqlite implementation that may
     // change in the future.
-    let sqlQuery = "SELECT url, title, frecency, " +
-                   "       last_visit_date as lastVisitDate, favicon, mime_type, " +
-                   "       \"history\" as type " +
-                   "FROM " +
-                   "(    " +
-                   " SELECT rev_host, moz_places.url, moz_favicons.data as favicon, mime_type, title, frecency, last_visit_date " +
-                   " FROM moz_places " +
-                   " LEFT JOIN moz_favicons " +
-                   " ON favicon_id = moz_favicons.id " +
-                   " WHERE hidden = 0 AND last_visit_date NOTNULL " +
-                   " ORDER BY rev_host, frecency, last_visit_date, moz_places.url DESC " +
-                   ")    " +
-                   "GROUP BY rev_host " +
-                   "ORDER BY frecency DESC, lastVisitDate DESC, url " +
-                   "LIMIT :limit";
+    let sqlQuery = `SELECT url, title, frecency,
+                          last_visit_date as lastVisitDate, favicon, mime_type,
```

Comment:
Nit: this seems a bit confusing that we're mixing some `lastVisitDate` camelCase with `mime_type` underscores. Not sure if it's worth standardizing on one vs the other [in a future PR].